quinn-h3/src/client.rs

```diff
@@ -173,10 +173,24 @@ impl SendRequest {
             recv: None,
         }
     }
+
+    fn build_response(header: Header) -> Result<Response<()>, Error> {
+        let (status, headers) = header.into_response_parts()?;
+        let mut response = Response::builder();
+        response.status(status);
+        response.version(http::version::Version::HTTP_3);
+        *response
+            .headers_mut()
+            .ok_or_else(|| Error::peer("invalid response"))? = headers;
```

Comment:
When can this error arise?

Reply:
The [doc](https://docs.rs/http/0.1.18/http/request/struct.Builder.html#method.headers_mut) isn't clear about that, so I preferred not to `panic!` and didn't bother to write a more specific error.

I guess the error cases are: invalid method or uri.  These values are received from the server.

Reply:
When you call `headers_mut` you haven't yet passed in the headers, so they can't possibly be the cause of an error.

Reply:
You're right, this API is not meant to be used this way. I got both functions rewritten. They now emit the error on `.body()` call (the builder end), and then set headers in the real instance. Thanks for catching this.